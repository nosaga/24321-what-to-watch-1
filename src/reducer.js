//import films from "../../mocks/films.js";

const initialState = {
  genre: `All genres`,
};


const reducer = (state = initialState, action) => {


  if (action.type === `GENRE_CLICKED`) {
    console.log(state.genre);
  }
  return state;
  // switch(action.type) {
  //   case `GENRE_CLICKED`: return Object.assign({}, state,{
  //     newState.genre;
  //   });
  // }
}

export default reducer;
