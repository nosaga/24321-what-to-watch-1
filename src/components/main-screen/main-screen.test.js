import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import MainScreen from "./main-screen.jsx";

import {createStore} from "redux";
import {reducer} from "../../reducer";
import {Provider} from "react-redux";
const store = createStore(reducer);

Enzyme.configure({adapter: new Adapter()});
it(`MainScreen renders correctly`, () => {
  const wrapper = shallow(<Provider store={store}><MainScreen
    filmTitle = {``}/></Provider>);


  expect(wrapper).toMatchSnapshot();
});
