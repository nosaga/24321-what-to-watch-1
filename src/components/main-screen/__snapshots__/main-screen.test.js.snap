// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen renders correctly 1`] = `ShallowWrapper {}`;
